enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/ParameterSettings.tsx

```diff
@@ -0,0 +1,114 @@
+import { useDebouncedCallback } from "@mantine/hooks";
+import { useCallback, useMemo } from "react";
+import { P, match } from "ts-pattern";
+import { t } from "ttag";
+
+import { Stack, Text, TextInput } from "metabase/ui";
+
+import { useSdkIframeEmbedSetupContext } from "../context";
+import { useHideParameter } from "../hooks";
```

Comment:
Maybe this is my pet peeve. I don't like stuff that are created for a particular file to live further away. In this case, I would have either.
1. Put this hook right into this file, considering it's merely 100 LoC long OR
2. Put the hook in `ParameterSettings/hooks.ts` and the component in `ParameterSettings/ParameterSettings/hooks.tsx`, so they live in the same folder.

I'd only create a dedecated folder like this when it's apparent we need to use this function in more than a couple of places.

Reply:
Agree, fixed in https://github.com/metabase/metabase/pull/60233/commits/fce0da677fa36c1dc561379b4cf195a92dabcc6c. I went with option 2, so I can also move `ParameterVisibilityToggle` in the same folder so they live together.